java/core/src/main/java/sleeper/core/partition/PartitionTree.java

```diff
@@ -159,6 +160,54 @@ public Partition getNearestCommonAncestor(Partition a, Partition b) {
         return getRootPartition();
     }
 
+    public Stream<Partition> traverseLeavesFirst() {
+        return traverseLeavesFirst(getLeavesInTreeOrder(), new HashSet<>(), Stream.empty());
+    }
+
+    private List<Partition> getLeavesInTreeOrder() {
+        // Establish ordering of the leaves with a depth-first traversal.
```

Comment:
It would be worth extending this comment - is the point to produce a consistent ordering of the leaves, i.e. an ordering that will always be the same for this tree? If so, we should comment that it's the combination of the depth-first traversal plus the deterministic ordering of the children of a vertex that gives us that.

Reply:
I've added that, thanks very much.